File: dotnet/cafdemalihapa/Directories/Hosting/Feature/FrontEnd/FrontEndPrimary/FrontEndPrimaryDirectory.cs
using cafdemalihapa.Files;
using cafdemalihapa.Names;

namespace cafdemalihapa.Directories.Hosting.Feature.FrontEnd.FrontEndPrimary
{
    public class FrontEndPrimaryDirectory(
            FrontEndDirectory frontEndDirectory,
            PrimaryApplication primaryApplication
        )
    {
        public string GetName()
        {
            var frontEndDirectoryName = frontEndDirectory.GetName();
            var primaryApplicationName = primaryApplication.GetName();
            var name = $"{frontEndDirectoryName}.{primaryApplicationName}";
            return name;
        }

        public string GetPath()
        {
            var frontEndDirectoryPath = frontEndDirectory.GetPath();
            var name = GetName();

            var frontEndPrimaryDirectoryPath = Path.Combine(frontEndDirectoryPath, name);
            return frontEndPrimaryDirectoryPath;
        }

        public void Create()
        {
            var path = GetPath();
            if(!Directory.Exists(path))
            {
                Directory.CreateDirectory(path);
            }
        }

        public void CreateFiles()
        {
            var path = GetPath();
            FileService.CreateNumberedFiles(path);
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Hosting/Feature/FrontEnd/FrontEndPrimary/FrontEndPrimaryDirectory.fs
namespace cafdemwimu.console.Directories.Hosting.Feature.FrontEnd.FrontEndPrimary

open System.IO
open cafdemwimu.console.Files
open cafdemwimu.console.Names
open cafdemwimu.console.Directories.Hosting.Feature.FrontEnd

type FrontEndPrimaryDirectory(frontEndDirectory: FrontEndDirectory, primaryApplication: PrimaryApplication) =
    member _.GetName() =
        let frontEndDirectoryName = frontEndDirectory.GetName()
        let primaryApplicationName = primaryApplication.GetName()
        let name = $"{frontEndDirectoryName}.{primaryApplicationName}"
        name

    member this.GetPath() =
        let frontEndDirectoryPath = frontEndDirectory.GetPath()
        let name = this.GetName()
        let frontEndPrimaryDirectoryPath = Path.Combine(frontEndDirectoryPath, name)
        frontEndPrimaryDirectoryPath

    member this.Create() =
        let path = this.GetPath()
        if not (Directory.Exists(path)) then
            Directory.CreateDirectory(path) |> ignore

    member this.CreateFiles() =
        let path = this.GetPath()
        FileService.CreateNumberedFiles path
